utilities/rocAL/rocAL_unittests/pixel_comparison/image_comparison.py

```diff
@@ -0,0 +1,148 @@
+from PIL import Image
+import os
+import sys
+import datetime
+import logging
+
+def compare_pixels(img1, img2, aug_name, width, height, image_offset = 0):
+    pixel_difference = [0,0,0,0,0,0]
+    if'rgb' in aug_name:
+        pixels1 = img1.load()
+        pixels2 = img2.load()
+        total_valid_pixel_count = width * height * 3
+        for wt in range(width):
+            for ht in range(height):
+                ht = ht + image_offset
+                if pixels1[wt,ht] != pixels2[wt,ht]:
+                    r = abs(pixels1[wt,ht][0]-pixels2[wt,ht][0])
+                    g = abs(pixels1[wt,ht][1]-pixels2[wt,ht][1])
+                    b = abs(pixels1[wt,ht][2]-pixels2[wt,ht][2])
+                    if(r > 4):
+                        r = 5
+                    if(g > 4):
+                        g = 5
+                    if(b > 4):
+                        b = 5
+                    pixel_difference[r] += 1
+                    pixel_difference[g] += 1
+                    pixel_difference[b] += 1
+                else:
+                    pixel_difference[0] += 3
+    else:
+        pixels1 = img1.convert('L').load()
+        pixels2 = img2.convert('L').load()
+        total_valid_pixel_count = width * height * 1
+        for wt in range(width):
+            for ht in range(height):
+                ht = ht + image_offset
+                if pixels1[wt,ht] != pixels2[wt,ht]:
+                    pixel = abs(pixels1[wt,ht]-pixels2[wt,ht])
+                    if(pixel > 4):
+                        pixel = 5
+                    pixel_difference[pixel] += 1
+                else:
+                    pixel_difference[0] += 1
+    return pixel_difference, total_valid_pixel_count
+
+def main():
+    timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S_")
+    handlers = [logging.FileHandler('../log_file_' + timestamp + '.log'), logging.StreamHandler()]
+    logging.basicConfig(level = logging.INFO, handlers = handlers)
+    if( len(sys.argv) < 3 ):
+        logging.error('Pass atleast 2 arguments (python image_comparison.py <ref_output_folder_path> <rocal_ouput_folder_path>')
+        exit()
+
+    # Open the two images
+    ref_output_path = sys.argv[1]
+    rocal_output_path = sys.argv[2]
+
+    if not (os.path.exists(ref_output_path) and os.path.exists(rocal_output_path)):
+        logging.error("Path does not Exists")
+        exit()
+
+    total_case_count = 0
+    passed_case_count = 0
+    failed_case_count = 0
+    failed_case_list = []
+    golden_output_dir_list = os.listdir(ref_output_path)
+    rocal_output_dir_list = os.listdir(rocal_output_path)
+    randomized_augmentation = ["Snow", "Rain", "Jitter", "SNPNoise"]
```

Comment:
@fiona-gladwin what are these randomized augmentations? Why are they all set to pass? I though only resize is failing now

Reply:
These Augmentations generate random outputs for every run, Hence they are set to pass.